web/src/main/java/com/bazaarvoice/emodb/web/scanner/ScanUploader.java

```diff
@@ -96,33 +115,60 @@ private ScanPlan createPlan(String scanId, ScanOptions options) {
+        } catch (Exception e) {
+            _log.error("Failed to update the stash time for scan {}", scanId, e);
+            throw Throwables.propagate(e);
+        }
 
-            // Notify the workflow that the scan can be started
-            _scanWorkflow.scanStatusUpdated(scanId);
+        // spawn a thread and do the below asynchronously as we would have to wait for a minute to continue to scan and we don't to include that delay in here for responding.
+        new Thread(() ->
+        {
+            try {
+                // We would like to wait 1 minute here to continue to scan to make sure scan don't miss any deltas that are written before the compaction control time.
+                Thread.sleep(_compactionControlBufferTimeInMillis);
 
-            // Send notification that the scan has started
-            _stashStateListener.stashStarted(status.asPluginStashMetadata());
-        } catch (Exception e) {
-            _log.error("Failed to start scan and upload for scan {}", scanId, e);
+                // Create the scan
+                _scanStatusDAO.updateScanStatus(status);
+               scanCreated.set(true);
+
+                // Notify the workflow that the scan can be started
+                _scanWorkflow.scanStatusUpdated(scanId);
 
-            if (scanCreated) {
-                // The scan was not properly started; cancel the scan
+                // Send notification that the scan has started
+                _stashStateListener.stashStarted(status.asPluginStashMetadata());
+            } catch (Exception e) {
+                _log.error("Failed to start scan and upload for scan {}", scanId, e);
+
+                // Delete the entry of the scan start time in Zookeeper.
                 try {
-                    _scanStatusDAO.setCanceled(scanId);
-                } catch (Exception e2) {
-                    // Don't mask the original exception but log it
-                    _log.error("Failed to mark unsuccessfully started scan as canceled: [id={}]", scanId, e2);
+                    _compactionControlSource.deleteStashTime(scanId, _dataCenters.getSelf().getName());
+                } catch (Exception ex) {
+                    _log.error("Failed to delete the stash time for scan {}", scanId, ex);
                 }
-            }
 
-            throw Throwables.propagate(e);
-        }
+                if (scanCreated.get() == true) {
+                    // The scan was not properly started; cancel the scan
+                    try {
+                        _scanStatusDAO.setCanceled(scanId);
+                    } catch (Exception e2) {
+                        // Don't mask the original exception but log it
+                        _log.error("Failed to mark unsuccessfully started scan as canceled: [id={}]", scanId, e2);
+                    }
+                }
+                throw Throwables.propagate(e);
```

Comment:
No point throwing this exception; it's in a thread with no Emo code to catch it.